Add documentation for setting up CRI-O with Flatcar

Aakash788 opened the pull request:
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Document the process of installing CRI-O using systemd-sysexts on Flatcar Container Linux. This guide provides a straightforward and supported method for setting up CRI-O, making it easier for users to leverage this container runtime on Flatcar.

Closes #8172.

#### Which issue(s) this PR fixes:

Add installation guide for CRI-O on Flatcar Container Linux, utilizing the systemd-sysexts feature. There was a gap in the existing documentation, and this aims to fill that by providing clear, step-by-step instructions.

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```
None
```

haircommander commented:
/ok-to-test

thanks @Aakash788 ! there are some [markdown lints](https://github.com/cri-o/cri-o/actions/runs/9942723595/job/27467292846?pr=8381) to address

Aakash788 commented:
Actually I thought they were just a warning. I'll fix them quickly. And anything else should I add or remove please, I am open for suggestions.

haircommander commented:
generally it LGTM!

kwilczynski commented:
@Aakash788, please fix your Git commit message to match the subject here. Thank you.

kwilczynski commented:
/release-note-none

Aakash788 commented:
Hii @mrunalp i was thinking that if we make a INSTALL-cri-o.sh. i mean ya we can follow the installation guide but  wouldnt it be good if we have a script which can install cri-o in any system (or install cri-o from package manager) and also install cri-o at multiple system at the same time with ansible. (and that script available on cri-o website os just download and run it.)

haircommander commented:
interesting idea @Aakash788 ! This is similar to our [get script](https://github.com/cri-o/packaging/blob/main/get) but there are a bunch of different ways to install. I think traditionally we've shown all the different ways and let users choose, but if you think having a canoncial script to do it would be useful, we would likely accept such a suggestion. maybe this should be a follow-up? WDYT @cri-o/cri-o-maintainers

Aakash788 commented:
it is showing that in `verify / space-at-eol` i had left a white space in line 133 but there is no unnecessary space at Line 133 in install.md (did i did something wrong and still not able to figure out?? 😅)

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+
+- Flatcar Linux installed on your machine. [(Flatcare Installation guide)](https://www.flatcar.org/docs/latest/installing/)
+- Systemd extensions (sysexts) enabled on Flatcar Linux. [(Enable Sysext in Flatcar)](https://www.flatcar.org/docs/latest/provisioning/sysext/)
+- `Curl` or `wget` for downloading files.
+- `Tar` for extracting the CRI-O binaries.
+- `sed` for editing files in-place.
```
```suggestion
- `curl` or `wget` for downloading files.
- `tar` for extracting the CRI-O binaries.
- `sed` for editing files in-place.
```

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+- `Curl` or `wget` for downloading files.
+- `Tar` for extracting the CRI-O binaries.
+- `sed` for editing files in-place.
+- sufficient privileges (using sudo if necessary).
```
```suggestion
- Sufficient privileges (using sudo if necessary).
```

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
 
+## Install CRI-O on Flatcar with Sysexts
+
+Installing CRI-O on Flatcar Linux with support for systemd extensions (sysexts),
```
```suggestion
Installing CRI-O on Flatcar Container Linux with support for systemd extensions (sysexts),
```

kwilczynski replied:
We would use either "Flatcar" or "Flatcar Container Linux", if possible, and as appropriate.

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+
+### Prerequisites
+
+- Flatcar Linux installed on your machine. [(Flatcare Installation guide)](https://www.flatcar.org/docs/latest/installing/)
```
```suggestion
- Flatcar Linux installed on your machine. [(Flatcar Installation guide)](https://www.flatcar.org/docs/latest/installing/)
```

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+To install CRI-O on Flatcar Linux with sysexts, follow these steps:
+
+- Step 1: Download the script
+Sample Extension script of install cri-o using sysext in [here](https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh)
```
```suggestion
Sample Extension script of install CRI-O using sysext in [here](https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh)
```

kwilczynski commented on install.md:
````diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+  - Using Curl:
+
+  ```bash
+  curl -L https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh
+  -o create_crio_sysext.sh
````
Why?

```suggestion
  curl -L https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh -o create_crio_sysext.sh
```

Aakash788 replied:
MD013/line-length error, line length should in under 80 char if i do it in suggested way it will be 85

kwilczynski replied:
Right. This is silly then. Especially since there are things that can be easily wrapped in markdown.

Aakash788 replied:
Sorry. But i couldn't get what you are suggesting up there in the last reply

```bash
curl -L \
https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh \
-o create_crio_sysext.sh
```

it is better this way ???

kwilczynski replied:
Looks better. And in terms of my reply. I was simply being grumpy about silly linter rules.

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+- Step 1: Download the script
+Sample Extension script of install cri-o using sysext in [here](https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh)
+
+  - Using Curl:
```
```suggestion
  - Using curl:
```

kwilczynski commented on install.md:
````diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+  -o create_crio_sysext.sh
+  ```
+
+  - Using Wget:
````
```suggestion
  - Using wget:
```

kwilczynski commented on install.md:
````diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+  wget -O create_crio_sysext.sh https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh
+  ```
+
+Make the Script Executable
````
```suggestion
Make the script executable
```

kwilczynski commented on install.md:
```diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+
+- Step 2: Run the Script
+
+Execute the script with the required arguments:
+The version of CRI-O you wish to install.
+And Also the name you wish to give to the sysext image.
+Optionally, set the `ARCH` environment variable if you need a specific architecture.
```
Was this intended to be a bullet list?

kwilczynski commented on install.md:
````diff
@@ -51,6 +55,101 @@ If there's a version or operating system that is missing, please [open an issue]
+```
+
+- Step 3: Deploy the System Extension
+  -Once the script completes, you will have a `.raw` sysext image file named as
+  per your `SYSEXTNAME` argument.
````
```suggestion
  - Once the script completes, you will have a `.raw` sysext image file named as
  per your `SYSEXTNAME` argument.
```

kwilczynski commented:
@Aakash788, before you submit something, please re-read it at least once yourself. Fix any typos, keep things capitalised consistently, add missing full stops, etc.

Aakash788 commented:
i may have made a big mistake (actually after this [64d6f02](https://github.com/cri-o/cri-o/pull/8381/commits/64d6f020f43d2c0ac938bc8b68b5fbc27f58ab15) i just missed one little thing but i was not able to revert it)but now i have created a big mess.

kwilczynski commented:
> i may have made a big mistake (actually after this [64d6f02](https://github.com/cri-o/cri-o/pull/8381/commits/64d6f020f43d2c0ac938bc8b68b5fbc27f58ab15) i just missed one little thing but i was not able to revert it)but now i have created a big mess.

@Aakash788, what happened? You lost some changes/work?

Check your git reflog, there should be things there, in a case you need to recover some work.

Aakash788 commented:
no  not like that, @kwilczynski  actully (i feel so dumb😅) right now from last commit everything this complete. but after [64d6f02](https://github.com/cri-o/cri-o/pull/8381/commits/64d6f020f43d2c0ac938bc8b68b5fbc27f58ab15) commit i I realised the indentation was inaccurate  in step 2 so just wanted to adjust (then i thought i should remove last commit then add all of that in one final and not able to because HEAD was change to fix/senitize-user-agent) i just wanted to make as less commit as possible and in the end added more 3 commits 😅

kwilczynski commented:
> no not like that, @kwilczynski actully (i feel so dumb😅) right now from last commit everything this complete. but after [64d6f02](https://github.com/cri-o/cri-o/pull/8381/commits/64d6f020f43d2c0ac938bc8b68b5fbc27f58ab15) commit i I realised the indentation was inaccurate in step 2 so just wanted to adjust (then i thought i should remove last commit then add all of that in one final and not able to because HEAD was change to fix/senitize-user-agent) i just wanted to make as less commit as possible and in the end added more 3 commits 😅

@Aakash788, ah OK. So just wanted to rebase and something went wrong... I hope you haven't lost your work. :smile:

kwilczynski commented on install.md:
````diff
@@ -49,7 +53,103 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+chmod +x create_crio_sysext.sh
+```
+
+- Step 2: Run the script.
````
```suggestion
- Step 2: Run the installation script:
```

kwilczynski commented on install.md:
````diff
@@ -49,7 +53,103 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+
+  - Using wget:
+
+    ```bash
+    wget -O create_crio_sysext.sh https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh
+    ```
````
```suggestion
    ```bash
    wget https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh \
    -O create_crio_sysext.sh
    ```
```

kwilczynski commented on install.md:
```diff
@@ -49,7 +53,104 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+
+To install CRI-O on Flatcar Container Linux with sysexts, follow these steps:
+
+- Step 1: Download the installation script:
```
```suggestion
- Step 1: Download the installation script:  
```

kwilczynski commented on install.md:
````diff
@@ -49,7 +53,104 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+  - Using wget:
+
+    ```bash
+    wget https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh \
+    -O create_crio_sysext.sh
````
```suggestion
    wget \
    https://github.com/flatcar/sysext-bakery/blob/main/create_crio_sysext.sh \
    -O create_crio_sysext.sh
```

kwilczynski commented on install.md:
````diff
@@ -49,7 +53,105 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+  ./create_crio_sysext.sh 1.28.4 crio-sysext
+  ```
+
+- Step 3: Deploy the system extension.
````
```suggestion
- Step 3: Deploy the system extension:
```

kwilczynski commented on install.md:
````diff
@@ -49,7 +53,105 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+  sudo systemctl start extensions-SYSEXTNAME.slice
+  ```
+
+- Step 4: Verify the installation.
````
```suggestion
- Step 4: Verify the installation:
```

kwilczynski commented on install.md:
```diff
@@ -49,7 +53,105 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+
+To install CRI-O on Flatcar Container Linux with sysexts, follow these steps:
+
+- Step 1: Download the installation script:  
```
```suggestion
- Step 1: Download the installation script:<br/>
```

kwilczynski commented on install.md:
```diff
@@ -49,7 +53,107 @@ of three minor releases. CRI-O also attempts to package generically for Debian
+
+- Step 1: Download the installation script:
+
+  Sample extension script for installing CRI-O using sysext in [here](
```
```suggestion
  Sample extension script for installing CRI-O using sysext is [here](
```

kwilczynski commented:
/approve
/lgtm

haircommander commented:
/approve

thanks @Aakash788

kwilczynski commented:
/retest

kwilczynski commented:
@cri-o/cri-o-maintainers, please have a look. This should be good to go.

saschagrunert commented:
@Aakash788 you need to dco-sign off the commit to continue here.

kwilczynski commented:
/hold

@Aakash788, can you squash commits into one? Thank you!

Aakash788 commented:
@saschagrunert i have removed all other commits (actually i signing off my commit and accidentally added other commits)
@kwilczynski  i squash all the commits into one is there anything left???

Aakash788 commented:
@saschagrunert i have removed all other commits (actually i signing off my commit and accidentally added other commits)
@kwilczynski  i squashed all the commits into one. is there any thing left??.(and actually i did lost all the work but i always keep a copy of my work in separate dir) but `git rebase` is cool too!

kwilczynski commented:
/approve 
/lgtm

kwilczynski commented:
/unhold
